File: src/clase-05/python/facade.video.lib.py
# Subsistema complejo (simulado)
class VideoReader: 
    def read(self, path): return f"raw({path})"

class VideoFilter:
    def denoise(self, raw): return f"denoised({raw})"

class VideoEncoder:
    def encode(self, stream, fmt): return f"encoded[{fmt}]({stream})"

# Facade
class VideoConverter:
    def encode(self, filename: str, target_format: str) -> str:
        reader = VideoReader()
        vf = VideoFilter()
        enc = VideoEncoder()
        raw = reader.read(filename)
        clean = vf.denoise(raw)
        return enc.encode(clean, target_format)

# Cliente
if __name__ == "__main__":
    api = VideoConverter()
    print(api.encode("cats.mov", "mp4"))  # encoded[mp4](denoised(raw(cats.mov)))

File: src/clase-05/typescript/facade.video.lib.ts
// Subsistema complejo (simulado)
class VideoReader { read(path: string) { return `raw(${path})`; } }
class VideoFilter { denoise(raw: string) { return `denoised(${raw})`; } }
class VideoEncoder { encode(stream: string, fmt: string) { return `encoded[${fmt}](${stream})`; } }

// Facade
class VideoConverter {
  encode(filename: string, targetFormat: string): string {
    const reader = new VideoReader();
    const filter = new VideoFilter();
    const encoder = new VideoEncoder();
    const raw = reader.read(filename);
    const clean = filter.denoise(raw);
    return encoder.encode(clean, targetFormat);
  }
}

// Cliente
const api = new VideoConverter();
console.log(api.encode("cats.mov", "mp4"));
